src/loader.ts

```diff
@@ -0,0 +1,25 @@
+import type { LoadFnOutput, LoadHookContext } from "node:module";
+import { transformSync } from "./index.ts";
+
+type NextLoad = (
+	url: string,
+	context?: LoadHookContext,
+) => LoadFnOutput | Promise<LoadFnOutput>;
+
+export async function load(
+	url: string,
+	context: LoadHookContext,
+	nextLoad: NextLoad,
+) {
+	const { format } = context;
+	if (format.includes("typescript")) {
+		const { source = "" } = await nextLoad(url, context);
```

Comment:
Wouldn't `nextLoad` already have stripped the types if you're using `--experimental-strip-types`?

Reply:
yes should been fixed now